.github/workflows/build_wheels.yml

```diff
@@ -25,7 +25,7 @@ jobs:
       - name: Build wheels
         uses: pypa/cibuildwheel@v2.1.1
         env:
-          CIBW_BEFORE_BUILD: pip install cython
+          CIBW_BEFORE_BUILD: pip install cython numpy
```

Comment:
Why do we need to install `numpy` explicitly? It's already specified as a dependency in `setup.py`.

Reply:
Since it a depency of `setup.py` we need to install it before we run `setup.py` so that it can pass the correct numpy path to cmake. When did install numpy explictly I got a import error. Before this `setup.py` didn't need numpy because we weren't using Cython's numpy API.

Reply:
I think this can be solved properly by adding a `pyproject.toml`.

Reply:
For this PR, I think we should only add a minimal `pyproject.toml`. It would also be a good starting point in migrating from the (more or less) deprecated `setup.py` to `pyproject.toml/setup.cfg` build format later on.

This should suffice (with a drive by fix of adding Cython as depency):

```
[build-system]
requires = ["cython", "numpy", "setuptools"]
```

This minimal file (+ changes to the build commands in CI-configuration) would at least eliminate the need to install numpy explicitely, where we don't call `setup.py` with additional parameters. To trigger the installation of build requirements, we have to use PEP517 conform calling mechanisms (like `pip`, `wheel` or `build`) and remove explicit calls to `setup.py build_ext`.

Example (`source-distribution` job in `build_wheels.yml`):

```
#Before
pip install cython numpy
python3 setup.py build_ext
python3 setup.py sdist

# After
pip install build
python3 -m build --sdist .
```

For other cases like `python setup.py build_ext --inplace --external-tlx=...` we have to write a wrapper in order to pass custom options to `pip install`. However, I would propose to leave this out of this PR.